src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -1582,52 +1592,71 @@ SubqueryPushdownMultiNodeTree(Query *queryTree)
 static List *
 FlattenJoinVars(List *columnList, Query *queryTree)
 {
-	ListCell *columnCell = NULL;
-	List *rteList = queryTree->rtable;
 	List *flattenedExprList = NIL;
+	ListCell *columnCell = NULL;
+	FlattenJoinVarsMutatorContext context = { queryTree };
 
 	foreach(columnCell, columnList)
 	{
-		Var *column = (Var *) lfirst(columnCell);
-		RangeTblEntry *columnRte = NULL;
-		PlannerInfo *root = NULL;
+		Node *column = strip_implicit_coercions(
```

Comment:
Do we need `strip_implicit_coercions ()` here as well? What is your thought on my [comment](https://github.com/citusdata/citus/pull/2631#discussion_r268566482)? I think if we do that, we shouldn't need this as well.

Reply:
we need `strip_jmplicit_coercions` either before calling `FlattenJoinVarsMutator` or after. 
Problem is with `some_var::int8` type casting. It appears as a function. If we provide the node the mutator function we got function expression back with inner var mutated (if needed). So we still need to do `strip_jmplicit_coercions` 

I felt like doing it afterwards. I have no strong opinion against doing it before.

Reply:
Addition, I still do copyObject after strip....

Reply:
> What is your thought on my comment? 

What I'm saying is that `expression_tree_mutator ` doesn't call mutate function on `Var`s. Don't you think that's a problem?

Reply:
we handle top level var's in the function. So mutator function is not called with Var node. For everything else it does mutating of the node via FLATCOPY and MUTATE on the child.

we should be fine here.

Reply:
Just noting for future readers, we had a quick call with @mtuncer and he agreed to change the call to `FlattenJoinVarsMutator ()`, so marking as resolved.